src-tauri/src/aux_functions/graph_init.rs

```diff
@@ -53,7 +55,25 @@ pub fn load_graph(packets: Vec<NeighborInfo>, mut loc_hashmap: HashMap<u32, Mesh
                 neighbor_position.longitude_i,
                 neighbor_position.altitude,
             );
-            let radio_quality = neighbor.snr;
+            // If we have previously seen the snr for this edge, use it if its timestamp is more recent
+            let mut radio_quality = neighbor.snr;
+            let edge_exists = snr_hashmap.contains_key(&(neighbor.id, packet.id));
+            if edge_exists {
+                let alternate_snr = snr_hashmap.get_mut(&(neighbor.id, packet.id)).unwrap().0;
+                let alternate_timestamp = snr_hashmap.get_mut(&(neighbor.id, packet.id)).unwrap().1;
+                if neighbor.timestamp >= alternate_timestamp {
+                    // Use the new snr
+                    snr_hashmap
+                        .insert((neighbor.id, packet.id), (neighbor.snr, neighbor.timestamp));
+                } else {
+                    // Use the old snr
+                    radio_quality = alternate_snr;
+                }
+            } else {
+                // Because our node will be the next node's neighbor, add it to the hashmap in reverse order
+                snr_hashmap.insert((packet.id, neighbor.id), (neighbor.snr, neighbor.timestamp));
+            }
+            // Add edge to lists
             edge_left_endpoints.push(node_index);
```

Comment:
I am just realizing this but this bit would create parallel edges since you are never checking if the edge already exists

Reply:
It doesn't matter to functionality since our graph can't really handle parallel edges but edge_idx_map of graph would still keep track of these.

Reply:
it does, but should be fine. I could do the work to remove parallel edges here -- it might make things clearer. This whole function is also a monolithic mess, so I could clarify that. 

I left it because this module began as a wrapper that didn't make very many decisions, so I assumed that the graph creation itself would make decisions about which info to include/how to normalize info.

Reply:
So it doesn't cause any problems because as I said our algorithms and graph can't really handle parallel edges and these are automatically handled like you said. But the edge_idx map field would still store these parallel edges, and it's just a matter of memory overhead which is definitely not a significant problem because our graphs are really small, but it'd be great if you could take that into consideration just to be safe and future-proof!

Reply:
Also I should note that the reason why I coded our graph in such a way that it would "technically" allow parallel edges is because in the future we might decide to make this a possibility and abandon petgraph.

Reply:
Updated `load_graph` to only load a single edge for each pair of nodes. It should be an easy matter to change that to double edges if we ever want to support parallel edges with different weights. For now (and for simplicity's sake), we're loading a single edge.